python/delta_sharing/rest_client.py

```diff
@@ -162,10 +167,19 @@ def close(self):
         self._session.close()
 
     def _get_internal(self, target: str, data: Optional[Dict[str, Any]] = None):
-        return self._request_internal(request=self._session.get, target=target, data=data)
+        return self._request_internal_with_backoff(
+            request=self._session.get, target=target, data=data
+        )
 
     def _post_internal(self, target: str, data: Optional[Dict[str, Any]] = None):
-        return self._request_internal(request=self._session.post, target=target, data=data)
+        return self._request_internal_with_backoff(
+            request=self._session.post, target=target, data=data
+        )
+
+    def _request_internal_with_backoff(self, request, target: str, data: Optional[Dict[str, Any]]):
+        return self._retry_with_exponential_backoff(
+            lambda: self._request_internal(request=request, target=target, data=data)
+        )
```

Comment:
After a second look, I found this is not a good idea.

As `self._request_internal(request=request, target=target, data=data)` returns a generator, the `_retry_with_exponential_backoff` is only covering the creation of the generator which even doesn't trigger the request yet.

We should pass the whole process to it, e.g., https://github.com/delta-io/delta-sharing/blob/14336db83ad5aba81d4f2b18ad410d20a36e15bd/python/delta_sharing/rest_client.py#L155-L164

Reply:
Good point. Do we want to use a decorator in this case?

Reply:
Do you mean decorating functions like `list_shares`, `list_schemas`, etc. with it?
That sounds good.

Reply:
Decorated list_shares, list_schemas, etc with a retry decorator. Can you take another look? Thanks!